File: backend/src/trotamundos/lrembecki.core.trotamundos/Pages/AboutUs/AboutUsVM.cs
using lrembecki.core.storage;

namespace lrembecki.core.trotamundos.Pages.AboutUs;

public record AboutUsVM(
    string Title,
    string Description,
    string FooterDescription,
    string FooterHighlight,
    Guid ImageId,
    StorageVM Image,
    List<AboutUsItemVM> Items,
    List<AboutUsPersonVM> Persons
)
{
    internal static AboutUsVM Map(AboutUsEntity entity)
    {
        if (entity == null) return null!;

        return new(
            entity.Title,
            entity.Description,
            entity.FooterDescription,
            entity.FooterHighlight,
            entity.ImageId,
            StorageVM.Map(entity.Image),
            [.. entity.Items.Select(AboutUsItemVM.Map)],
            [.. entity.Persons.Select(AboutUsPersonVM.Map)]
        );
    }

    public static AboutUsVM Empty => new(
        Title: string.Empty,
        Description: string.Empty,
        FooterDescription: string.Empty,
        FooterHighlight: string.Empty,
        ImageId: new Guid(),
        Image: null!,
        Items: [],
        Persons: []
    );

}

File: frontend/src/app/modules/trotamundos/about-us/about-us.vm.ts
import { StorageVM } from '@app/shared/storage/storage.vm';

export class AboutUsItemVM {
  order: number = null!;
  icon: string = null!;
  title: string = null!;
  description: string = null!;

  constructor(init?: Partial<AboutUsItemVM>) {
    Object.assign(this, init);
  }
}

export class AboutUsPersonVM {
  order: number = null!;
  name: string = null!;
  description: string = null!;
  imageId: string = null!;
  image: StorageVM = new StorageVM();

  constructor(init?: Partial<AboutUsPersonVM>) {
    Object.assign(this, init);
    this.image = new StorageVM(init?.image);
  }
}
export class AboutUsVM {
  id: string = null!;
  title: string = null!;
  description: string = null!;
  footerDescription: string = null!;
  footerHighlight: string = null!;
  imageId: string = null!;
  image: StorageVM = new StorageVM();
  items: AboutUsItemVM[] = [];
  persons: AboutUsPersonVM[] = [];

  constructor(init?: Partial<AboutUsVM>) {
    Object.assign(this, init);
    this.image = new StorageVM(init?.image);
    this.items = init?.items?.map((item) => new AboutUsItemVM(item)) ?? [];
    this.persons =
      init?.persons?.map((person) => new AboutUsPersonVM(person)) ?? [];
  }
}
